lib/ocrypto/ec_key_pair.go

```diff
@@ -0,0 +1,99 @@
+package ocrypto
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+)
+
+type ECCMode uint8
+
+const (
+	ECCModeSecp256r1 ECCMode = 0
+	ECCModeSecp384r1 ECCMode = 1
+	ECCModeSecp521r1 ECCMode = 2
+	ECCModeSecp256k1 ECCMode = 3
+)
+
+type ECKeyPair struct {
+	PrivateKey *ecdsa.PrivateKey
+}
+
+// NewECKeyPair Generates an EC key pair of the given bit size.
+func NewECKeyPair(mode ECCMode) (ECKeyPair, error) {
```

Comment:
Should we just take in a type [Curve](https://pkg.go.dev/crypto/elliptic#Curve) here?

Reply:
The mode is used in the nano header, so while it would be fine to use a curve on this call, we'd still need to convert back and forth between curve and enum values at some point.  Seems simpler to be consistent and use the enum everywhere.

Reply:
That avoids stuff like `NewECKeyPair(ModeToCurve(mode))` or  `cfg.bindingBody = eccMode((CurveToMode(c) >> 4) & 0x07)`